thavalon-webapp/src/utils/AccountManager.tsx

```diff
@@ -306,4 +319,84 @@ export class AccountManager {
             return httpResponse;
         });
     }
+
+    /**
+     * Creates a game made by the current user.
+     */
+    public async createGame(): Promise<HttpResponse> {
+        const httpResponse: HttpResponse = {
+            "result": true,
+            "message": ""
+        }
+
+        return await fetch("/api/add/game", {
+            method: "POST",
+            headers: {
+                "Content-Type": "application/json",
+                "Authorization": "Basic " + this.token,
+            },
+            credentials: "include"
+        }).then((response) => {
+            if (response.status === STATUS.OK) {
+                return response.json();
+            } else {
+                httpResponse.result = false;
+                httpResponse.message = "Unexpected return code from server: " + response.status;
+            }
+            return httpResponse;
+        }).then((createGameResponse: CreateGameResponse) => {
+            httpResponse.message = createGameResponse.friendCode;
+            return httpResponse;
+        }).catch((error) => {
+            console.log("Failed to create game, error is: " + error);
+            httpResponse.result = false;
+            httpResponse.message = "Unable to create game, try again";
+            return httpResponse;
+        });
+    }
+
+    /**
+     * Joins an existing game.
+     *
+     * @param friendCode the friend code for the game
+     * @param displayName the display name of the user joining the game
+     */
+    public async joinGame(friendCode: string, displayName: string): Promise<HttpResponse> {
+        const httpResponse: HttpResponse = {
+            "result": true,
+            "message": ""
+        }
+
+        const joinGameInfo: JoinGameInfo = {
+            "friendCode": friendCode,
+            "displayName": displayName,
+        }
+
+        return await fetch("/api/join/game", {
+            method: "POST",
+            body: JSON.stringify(joinGameInfo),
+            headers: {
+                "Content-Type": "application/json",
+                "Authorization": "Basic " + this.token,
+            },
+            credentials: "include"
+        }).then((response) => {
+            if (response.status === STATUS.OK) {
+                return response.json();
+            } else {
+                httpResponse.result = false;
+                httpResponse.message = "Unexpected return code from server: " + response.status;
+            }
+            return httpResponse;
```

Comment:
Doesn't this mean that the error message will always be "Unexpected return code from server: "? This error message displays to end users I think.

Reply:
hmm, guess we shouldn't show the actual rcode on the site, will console log it and have a generic error message saying try again

Reply:
Or we need an adaptor to generate a useful error message. It can probably be in my modal. For instance, an unregistered account should display a helpful error message, so should not entering the correct friend code. Not sure if this needs to be in your code or mine.

Reply:
We could have the server return a standardized set of error codes. Then `AccountManager` can pass them along to the UI code, which can render a message based on the code

Reply:
I'm ok with that, that makes sense to me - we can discuss more later and do this in a separate PR